plugin.py

```diff
@@ -1,149 +1,34 @@
-        return calculated == expected
-
-
-def unpack_server(zip_file):
-    if not is_valid_hash(zip_file, SERVER_HASH):
-        log_debug("Invalid hash, aborting")
-        return
-
-    target_dir = get_server_dir(SERVER_VERSION)
-    os.makedirs(target_dir, exist_ok=True)
-    log_debug("Unpacking server to: {}".format(target_dir))
-
-    with ZipFile(zip_file, "r") as f:
-        f.extractall(target_dir)
+from .server_zip_resource import ServerZipResource
 
-    # ZipFile removes permissions, make server executable
-    if not IS_WINDOWS:
-        os.chmod(os.path.join(target_dir, "language_server.sh"), 0o755)
-        os.chmod(os.path.join(target_dir, "launch.sh"), 0o755)
-
-    log_debug("Server installed")
-
-
-def download_server():
-    log_debug("Downloading server from {}".format(SERVER_URL))
-    target = sublime.active_window()
-    label = "Downloading Elixir language server binary"
-
-    with ActivityIndicator(target, label):
-        try:
-            tmp_file, _ = urlretrieve(SERVER_URL)
-            unpack_server(tmp_file)
-            os.unlink(tmp_file)
-        except Exception as ex:
-            log_debug("Failed downloading server: {}".format(ex))
-
-
-def download_server_async():
-    thread = Thread(target=download_server)
-    thread.start()
-
-
-def delete_server():
-    target_dir = get_server_dir(SERVER_VERSION)
-    log_debug("Deleting server from {}".format(target_dir))
-    shutil.rmtree(target_dir, ignore_errors=True)
-
-
-def is_server_downloaded():
-    return os.path.exists(get_server_exec())
-
-
-class LspElixirPlugin(LanguageHandler):
-    @property
-    def name(self):
-        return __package__.lower()
-
-    @property
-    def config(self):
-        config = {
-            "enabled": True,
-            "command": [get_server_exec()],
-        }
-
-        filename = '{}.sublime-settings'.format(__package__)
-        settings = sublime.load_settings(filename)
-        for key, default in DEAFULT_SETTINGS.items():
-            config[key] = settings.get(key, default)
+SERVER_URL = "https://github.com/elixir-lsp/elixir-ls/releases/download/v0.6.4/elixir-ls-1.11.zip"
+SERVER_HASH = "30df49d0c7c12717814a3b03d4b8064197820a10beac138ad6d63d28f5295a4f"
+SERVER_VERSION = "0.6.4"
 
-        return read_client_config(self.name, config)
+SERVER_EXECUTABLES = ["language_server.sh", "launch.sh"]
+BINARY_PATH = "language_server.bat" if sublime.platform() == 'windows' else "language_server.sh"
 
-    def on_start(self, window):
-        if not is_elixir_installed():
-            sublime.status_message(
-                "Please install Elixir for LSP-elixir plugin to work.")
-            return False
 
-        if not is_server_downloaded():
-            sublime.status_message(
-                "Elixir Language Server not yet downloaded.")
-            return False
+class LspElixirPlugin(GenericClientHandler):
+    package_name = __package__
+    __server = None
 
+    @classmethod
```

Comment:
Did the server name shown in the status change with this change. I think it will be `LSP-elixir` now while previously it was `lsp-elixir`. It's important to keep the old name since, for example, project overrides would not apply anymore otherwise. You can use https://sublimelsp.github.io/lsp_utils/client_handlers.html#lsp_utils.GenericClientHandler.get_displayed_name to override the name to lowercase.

Reply:
Right. 
![image](https://user-images.githubusercontent.com/1329716/108330567-d8b28f00-7221-11eb-9d99-fa38445a8b63.png)